common/schema/index.ts

```diff
@@ -244,6 +244,463 @@ function validateNoSpeakerOverlap(data: {
+ *    - Rationale: A clear position always overrides "unclear" classification
+ *
+ * 3. **Within-list deduplication**:
+ *    - Remove duplicates within each individual list
+ *    - First occurrence is kept, subsequent duplicates are dropped
+ *
+ * **Why use "agree version" for ambiguous speakers?**
+ * When moving an ambiguous speaker to no_clear_position, we use their data from the
+ * agree list. This is arbitrary but consistent - we need to pick one source for
+ * speaker metadata (name, strength annotations, etc.), and agree is checked first.
+ *
+ * **Examples**:
+ *
+ * ```typescript
+ * // Example 1: Ambiguous speaker
+ * Input:  { agree: ["1:Alice"], disagree: ["1:Alice"], no_clear_position: [] }
+ * Output: { agree: [], disagree: [], no_clear_position: ["1:Alice"] }
+ *
+ * // Example 2: Clear stance overrides unclear
+ * Input:  { agree: ["1:Alice"], disagree: [], no_clear_position: ["1:Alice"] }
+ * Output: { agree: ["1:Alice"], disagree: [], no_clear_position: [] }
+ *
+ * // Example 3: Multiple rules applied
+ * Input:  { agree: ["1:Alice", "2:Bob", "2:Bob"], disagree: ["1:Alice"], no_clear_position: ["2:Bob"] }
+ * Output: { agree: ["2:Bob"], disagree: [], no_clear_position: ["1:Alice"] }
+ * // Alice: ambiguous → no_clear | Bob: duplicates removed, clear stance (agree) kept
+ * ```
+ *
+ * @param data Raw crux data with potentially duplicate speakers
+ * @returns Reconciled crux data with each speaker appearing in exactly one list
+ * @public Exported for direct testing and reuse
+ */
+export function reconcileCruxSpeakers<
+  T extends {
+    agree: string[];
+    disagree: string[];
+    no_clear_position: string[];
+  },
+>(data: T): T {
+  // Track speaker metadata in single map (2-pass algorithm for optimal performance)
+  type SpeakerInfo = {
+    fullString: string; // Full speaker string (id:name or id:name | strength)
+    appearsIn: Set<"agree" | "disagree" | "no_clear">; // Which lists contain this speaker
+    countByList: { agree: number; disagree: number; no_clear: number }; // Appearances per list
+  };
+
+  const speakerMap = new Map<string, SpeakerInfo>();
+  let inputCount = 0;
+
+  // PASS 1: Build speaker map from all three lists
+  const processList = (
+    list: string[],
+    listType: "agree" | "disagree" | "no_clear",
+  ) => {
+    for (const speaker of list) {
+      const id = extractSpeakerId(speaker);
+      if (!id) continue; // Skip invalid/empty speakers
+
+      inputCount++;
+      const existing = speakerMap.get(id);
+      if (existing) {
+        existing.appearsIn.add(listType);
+        existing.countByList[listType]++;
+        // Preserve the richest speaker data (prefer longer strings or those with metadata like strength scores)
+        // Examples: "1:Alice | 0.9" is richer than "1:Alice", "1:Alice Smith" is richer than "1:Alice"
+        if (
+          speaker.length > existing.fullString.length ||
+          (speaker.includes("|") && !existing.fullString.includes("|"))
```

Comment:
This could prioritize a speaker with a longer name over a speaker with additional metadata.  If we want to prioritize keeping the strength score we could do something like this:


```suggestion
if (
  (speaker.includes("|") && !existing.fullString.includes("|")) ||
  (speaker.includes("|") === existing.fullString.includes("|") && speaker.length > existing.fullString.length)
)
```